samples/opengl/penguin/penguin.cpp

```diff
@@ -92,7 +92,9 @@ MyFrame::MyFrame(wxFrame *frame, const wxString& title, const wxPoint& pos,
 
     Show(true);
 
-    m_canvas = new TestGLCanvas(this, wxID_ANY, wxDefaultPosition,
+    wxGLAttributes attribList;
+    attribList.PlatformDefaults().Depth(16).EndList();
```

Comment:
Sorry, after actually looking at the code I don't understand why is this necessary -- the defaults already include `Depth(16)` on all platforms. So I think this change should be undone.

Reply:
I am afraid that is not true, this is what happens if I don't specify `Depth(16)` on Linux:
https://i.imgur.com/1f8yPCJ.png

This was the point of contention in the thread that I linked in the OP of this PR. Under Linux, wxWidgets fails to initialize the depth buffer properly.

Reply:
Ok upon further inspection of the documentation, I think I now realize that you were probably referring to `Defaults()`, and not `PlatformDefaults()` :sweat_smile:

Reply:
OK, this is due to passing `NULL` to `wxGLCanvas` ctor and `wxGLAttributes::AddDefaultsForWXBefore31()` not specifying depth in X11 implementation. The real fix is to stop using the old ctor taking `const int*` and pass it `wxGLAttributes::Defaults()` instead.

Could you please do it like this and check that it works?

Reply:
I think my confusion stemmed a little bit from the pyramid sample, which uses both
`vAttrs.PlatformDefaults().Defaults().EndList();` 
AND 
`vAttrs.PlatformDefaults().RGBA().DoubleBuffer().Depth(16).EndList();`

I guess I'll modify the penguin sample to use `PlatformDefaults().Defaults()`.

> The real fix is to stop using the old ctor taking const int* and pass it wxGLAttributes::Defaults() instead.
Are you referring to the cube sample? The code you highlighted in this review is from the penguin sample, which isn't using the integer array.

Reply:
> I think my confusion stemmed a little bit from the pyramid sample, which uses both `vAttrs.PlatformDefaults().Defaults().EndList();` AND `vAttrs.PlatformDefaults().RGBA().DoubleBuffer().Depth(16).EndList();`

The second one is clearly not ideal, there is no need to duplicate `Defaults()`. The first one is probably fine, even though I think it would make more sense to specify platform defaults after the global defaults and, in fact, they could be omitted entirely because I don't think the sample would need them and there are no platform-specific defaults at all right now anyhow.
 
> > The real fix is to stop using the old ctor taking const int* and pass it wxGLAttributes::Defaults() instead.
> 
> Are you referring to the cube sample? The code you highlighted in this review is from the penguin sample, which isn't using the integer array.

It does use it, see `TestGLCanvas` ctor in it: it calls `wxGLCanvas` ctor with `NULL`, but it should use the attributes object here instead. If you change this, everything should work without having to specify the depth explicitly.

Reply:
> It does use it, see `TestGLCanvas` ctor in it: it calls `wxGLCanvas` ctor with `NULL`, but it should use the attributes object here instead. If you change this, everything should work without having to specify the depth explicitly.

The original does not pass attributes, yes. My modified sample does. You're referring to the original, right? :sweat_smile: 

I have pushed another commit to the penguin sample, which uses `Defaults().PlatformDefaults()` now, please confirm if it is to your liking!